### ISSUE TITLE
Implement Cart Checkout (spec)

### ISSUE THREAD

**Issue by apbendi** [dcgtc/dgrants#94]
### Adding To Cart

- [x] When on the grant detail view, there should be an "Add to Cart" button
- [x] When viewing a list of grants, there should an Cart icon for add to cart
- [x] When either aforementioned button is clicked, the following data should be added to local storage:
     - [x] The grantId
     - [x] The donation token address (default to Dai)
     - [x] The donation token amount (default to 5 in raw decimal format)
- [x] There should be utility methods that abstract writing/reading cart data from local storage
- [x] The data should be returned in the following format:

#### Sample Schema

````js
[
    {
        grantId: 5,
        contributionTokenAddress: "0x6b175474e89094c44da98b954eedeac495271d0f",
        contributionAmount: "5000000000000000000",
    },
    {
        grantId: 7,
        contributionTokenAddress: "0xde30da39c46104798bb5aa3fe8b9e0e1f348163f",
        contributionAmount: "1337000000000000000",
    },
]
````

### Cart Page

- [x] The cart page should list the grants the user has added
    - [ ] It should load metadata for the grant via IPFS; dependent on #86 
    - [x] It should display the token and the human readable amount
- [x] The cart page should show a summarizations of how much of each token is being donated in total (e.g. "Donating 25 Dai + .4 ETH")
- [x] The cart page should allow the user to update their token and amount individually for each row
     - [x] The UI should list the following tokens in the UI (for now): DAI, ETH, USDC, GTC, UNI
     - [x] The list of supported tokens should use the [token list](https://uniswap.org/blog/token-lists/) format, and should be generalized to work with tokens of any name/decimals/symbols etc....
- [x] The cart page should allow the user to remove items from their cart
- [x] The cart page should allow users to clear their cart

### Additional Notes

* The implementation should tend toward @scco's [figma designs](https://www.figma.com/file/sUQSYyPfooqTcVdajfZm0t/dgrants?node-id=92%3A80), but do not need to match it precisely for now; more precise styling will be handled in a separate task
* We can also leave matching estimates and USD conversions out of the initial implementation and address them as separate features
